### DISCUSSION TITLE
How to explode CAD files to read in hidden entities

### DISCUSSION THREAD

**Discussion by Lukeor** [DomCR/ACadSharp#496] [Q&A]
Hello again @DomCR and thanks for your continued support in this library!

I have a query regarding some data that does not seem to be ready in by ACadSharp.

[Topographical Survey.zip](https://github.com/user-attachments/files/17871641/Topographical.Survey.zip)

In the file called **Topographical Survey - Original**, if I view the file in a CAD viewer such as DWG TrueView then I am able to see trees in the drawing:
![image](https://github.com/user-attachments/assets/499b761b-ab21-4cd5-a66d-cb5d76c80caa)

According to DWG TrueView, these trees belong to a layer called 'vegetation'.

However when reading in this CAD file with ACadSharp I am unable to obtain these trees. It seems that the vegetation layer contains numerous INSERT entities but that's it.

If I use AutoCAD to 'explode' the CAD drawing then I am able to read in these trees using ACadSharp. These trees now appear as POLYLINES in a layer called '0'.

Please see the file called **Topographical Survey - Exploded** to see the exploded version of the same CAD file. ACadSharp now reads in the trees within this file.

Is there any way to read in and access these trees (polylines) without having to explode the CAD file beforehand? Some users do not have access to AutoCAD and thus are unable to explode these files if need be.

Many thanks for your continued work and support!

Luke

**Comment by DomCR** [comment]
Hi @Lukeor,

I ran this code on my end:
```C#
	var trees = doc.Entities.Where(e => e is Insert insert 
		&& insert.Block.Name == "TREE_1"
		&& e.Layer.Name.Equals("VEGETATION", System.StringComparison.OrdinalIgnoreCase));
```

It got multiple entries with the correct layer and block:

![image](https://github.com/user-attachments/assets/b6c217e1-46c0-43f0-a522-11f3ff1536fe)

Can you provide an example of the code on your end? 
Which version of ACadSharp are you using?

Let me know if this helps.

**Reply by Lukeor** [reply]
Hi Dom,

I am using v2.4.0-beta of ACadSharp.

When searching within all block records contained in the file (ACadSharp finds 20 block records I believe) I then obtain around 1394 entities that belong to the vegetation layer.

I do this using the following code:
![image](https://github.com/user-attachments/assets/6a19aea7-d920-49a6-8a40-25d677ae4ba9)

All of these entities however are INSERT entities. I'm not particularly sure how I can use these INSERT features to then display/visualise the various trees that belong to the vegetation layer. As the trees consist of a canopy outline I was expecting the entities in the vegetation layer to be POLYLINES instead of INSERTS but that doesn't seem to be the case.

![image](https://github.com/user-attachments/assets/b76b99e9-d27d-45f1-9a4e-a2092605ffd3)

It's only when I explode the file in AutoCAD that I am now able to see the POLYLINE entities for the trees - although this time they belong to a layer called '0'.

Many thanks,
Luke

**Reply by DomCR** [reply]
There are new versions for ACadSharp available, is there a reason why you didn't update the package?

Please try to use the latest version, it seems to fix the issue on my end.

**Reply by Lukeor** [reply]
Hey @DomCR 

Apologies for the delay in getting back to you.

Nuget Packet Manager, which is what I use to manage the packages for my project, thinks that v2.4.0-beta is the latest version of ACadSharp available. However I have now manually switched over to v1.0.2.

I have managed to fix the issue with not being able to read in the trees from the CAD file. It turns out that I was not looking within the INSERT entities correctly in order to obtain the associated entities contained within the Block. A lack of understanding on my part!

Please accept my apologies for any waste of time of your part and thanks as always for the assistance.

Many thanks,
Luke

**Reply by DomCR** [reply]
Don't worry, is not a waste of time, this fixes is what makes the library improve.

I don't know why, but the nuget is badly sorted, having the suffix beta at the end should set the version as a pre-release, but it doesn't understand it, I've unlisted all pre-release versions to make sure that this doesn't happen.

Thanks for your support!
